tracer/src/Datadog.Trace/Ci/ITRClient.cs

```diff
@@ -0,0 +1,428 @@
+        {
+            return 0;
+        }
+
+        var remoteCommitsData = await SearchCommitAsync(localCommits).ConfigureAwait(false);
+        return await SendObjectsPackFileAsync(localCommits[0], remoteCommitsData).ConfigureAwait(false);
+    }
+
+    private async Task<string[]> SearchCommitAsync(string[] localCommits)
+    {
+        if (localCommits is null)
+        {
+            return null;
+        }
+
+        Log.Debug("ITR: Searching commits...");
+
+        var commitRequests = new CommitRequest[localCommits.Length];
+        for (var i = 0; i < localCommits.Length; i++)
+        {
+            commitRequests[i] = new CommitRequest(localCommits[i]);
+        }
+
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var jsonPushedSha = JsonConvert.SerializeObject(new DataArrayEnvelopeWithMeta<CommitRequest>(commitRequests, repository));
+        var jsonPushedShaBytes = Encoding.UTF8.GetBytes(jsonPushedSha);
+
+        return await WithRetries(InternalSearchCommitAsync, jsonPushedShaBytes, MaxRetries).ConfigureAwait(false);
+
+        async Task<string[]> InternalSearchCommitAsync(byte[] state, bool finalTry)
+        {
+            var request = _apiRequestFactory.Create(_searchCommitsUrl);
+            request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            Log.Debug("ITR: Searching commits from: {url}", _searchCommitsUrl.ToString());
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
+            var responseContent = await response.ReadAsStringAsync().ConfigureAwait(false);
+            if (response.StatusCode is < 200 or >= 300)
+            {
+                if (finalTry)
+                {
+                    try
+                    {
+                        Log.Error<int, string>("Failed to submit events with status code {StatusCode} and message: {ResponseContent}", response.StatusCode, responseContent);
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.Error<int>(ex, "Unable to read response for failed request with status code {StatusCode}", response.StatusCode);
+                    }
+                }
+
+                throw new WebException($"Status: {response.StatusCode}, Content: {responseContent}");
+            }
+
+            var deserializedResult = JsonConvert.DeserializeObject<DataArrayEnvelope<CommitResponse>>(responseContent);
+            if (deserializedResult.Data is null)
+            {
+                return null;
+            }
+
+            var stringArray = new string[deserializedResult.Data.Length];
+            for (var i = 0; i < deserializedResult.Data.Length; i++)
+            {
+                var value = deserializedResult.Data[i].Id;
+                if (ShaRegex.Matches(value).Count != 1)
+                {
+                    ThrowHelper.ThrowException($"The value '{value}' is not a valid Sha.");
```

Comment:
Is this behavior documented in the RFC?

Reply:
Nope, It's missing... we find out later that for security reasons we need to sanitize things we receive from the network because we are running a command in the target machine based on that, and can be use as an attack vector. I will update the RFC.